tests/mdf/test_mdf.py

```diff
@@ -86,21 +137,25 @@ def test_json_results_equivalence(
     input_dict_str,
     simple_edge_format,
 ):
+    comp_inputs = {composition_name: input_dict_str}
+
     # Get python script from file and execute
-    filename = os.path.join(os.path.dirname(__file__), filename)
-    with open(filename, 'r') as orig_file:
-        exec(orig_file.read())
-        exec(f'{composition_name}.run(inputs={input_dict_str})')
-        orig_results = eval(f'{composition_name}.results')
+    orig_script = read_defined_model_script(filename)
+    orig_results, orig_globals, orig_locals = get_model_results_and_state(
+        orig_script, comp_inputs
+    )
 
     # reset random seed
     pnl.core.globals.utilities.set_global_seed(0)
     # Generate python script from JSON summary of composition and execute
-    json_summary = pnl.generate_json(eval(f'{composition_name}'), simple_edge_format=simple_edge_format)
-    exec(pnl.generate_script_from_json(json_summary))
-    exec(f'{composition_name}.run(inputs={input_dict_str})')
-    new_results = eval(f'{composition_name}.results')
-    assert pnl.safe_equals(orig_results, new_results)
+    json_summary = pnl.generate_json(
```

Comment:
SHould this use `get_mdf_serialized`? There's a warning that `generate_json` is deprecated.